### DISCUSSION TITLE
Deploy app on Railway reports success, although domain return timeout exceeded after a ~1 min loading

### DISCUSSION THREAD

**Discussion by Or3yn** [community/community#131500] [Programming Help] [answered]
### Railway deploy. Timeout exceeded while success deploy check

How to ship data correctly to my domain?

I will just describe everything i tried and give all possible info below. There might be extra info, sorry for that. I have 0 idea of a problem's root.

-------------------------
Here's my domain from railway: 
https://football-fantasy-production.up.railway.app/
------------------------------------------
Repository with entire code:
https://github.com/Or3yn/football-fantasy-client

Quick descriptiion:

 it's vite+swc+ts app (basic build) with railway config(railway.toml), also i configured building from github repository (without Dockerfile), so it works via nixpacks. In railway workflow i've created 2 gitgub servers, -->this and for rOr app which is deployed too(i loaded it from template and didn't touch it). 3rd server - potgresql. I tried to bond all of them just a little.
 
 
 i have vite config:
 
import { defineConfig, loadEnv } from 'vite';
import react from '@vitejs/plugin-react-swc';
import path from 'path';

export default defineConfig(({ mode }) => {
  const envDir = path.resolve(__dirname, "environment");
  const env = loadEnv(mode, envDir, '');
  console.log('Loaded environment variables:', env);

  return {
    plugins: [react()],
    base: './',
    server: {
      host: '0.0.0.0',
      strictPort: true,
      open: false,
      port: 8080,
    },
    build: {
      outDir: 'dist',
      assetsDir: '',
    },
    define: {
      __APP_ENV__: JSON.stringify(env.APP_ENV),
      __VITE_API_URL__: JSON.stringify(env.VITE_API_URL),
      __VITE_API_KEY__: JSON.stringify(env.VITE_API_KEY),
    },
  };
});

here's toml:

[build]
builder = "nixpacks"
buildCommand = "yarn install && tsc -b && vite build"

[deploy]
restartPolicyMaxRetries = 10
startCommand = "npx serve -s dist"
healthcheckPath = "/ping"
healthcheckTimeout = 100
restartPolicyType = "ON_FAILURE"

[env]
NIXPACKS_PATH = "dist"  # Укажите необходимое значение для переменной
----------------------------------------------
 railway logs
 
Starting Container
npm WARN config production Use `--omit=dev` instead.
 INFO  Accepting connections at http://localhost:8080
 HTTP  7/9/2024 7:39:44 AM 100.64.0.2 GET /ping
 HTTP  7/9/2024 7:39:44 AM 100.64.0.2 Returned 200 in 22 ms
----------------
Project: genuine-miracle
Environment: production
Service: football-fantasy
----------------------------
  VITE v5.3.3  ready in 1150 ms

  ➜  Local:   http://localhost:8080/
  ➜  Network: http://10.255.255.254:8080/
  ➜  Network: http://172.27.205.145:8080/

It works only locally.
------------------------------------------------------
root@INBOOKX2PLUS:~# curl http://172.27.205.145:8080/
<!doctype html>
<html lang="en">
  <head>
    <script type="module">import { injectIntoGlobalHook } from "/@react-refresh";
injectIntoGlobalHook(window);
window.$RefreshReg$ = () => {};
window.$RefreshSig$ = () => (type) => type;</script>

    <script type="module" src="/@vite/client"></script>

    <meta charset="UTF-8" />
    <link rel="icon" type="image/svg+xml" href="/vite.svg" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Vite + React + TS</title>
  </head>
  <body>
    <div id="root"></div>
    <script type="module" src="/src/main.tsx"></script>
  </body>
</html>
root@INBOOKX2PLUS:~# curl http://localhost:8080/ping
<!doctype html>
<html lang="en">
  <head>
    <script type="module">import { injectIntoGlobalHook } from "/@react-refresh";
injectIntoGlobalHook(window);
window.$RefreshReg$ = () => {};
window.$RefreshSig$ = () => (type) => type;</script>

    <script type="module" src="/@vite/client"></script>

    <meta charset="UTF-8" />
    <link rel="icon" type="image/svg+xml" href="/vite.svg" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Vite + React + TS</title>
  </head>
  <body>
    <div id="root"></div>
    <script type="module" src="/src/main.tsx"></script>
  </body>
</html>
------------------------------------------------------
root@INBOOKX2PLUS:~# curl http://172.27.205.145:8080/
<!doctype html>
<html lang="en">
  <head>
    <script type="module">import { injectIntoGlobalHook } from "/@react-refresh";
injectIntoGlobalHook(window);
window.$RefreshReg$ = () => {};
window.$RefreshSig$ = () => (type) => type;</script>

    <script type="module" src="/@vite/client"></script>

    <meta charset="UTF-8" />
    <link rel="icon" type="image/svg+xml" href="/vite.svg" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Vite + React + TS</title>
  </head>
  <body>
    <div id="root"></div>
    <script type="module" src="/src/main.tsx"></script>
  </body>
</html>
---------------------------------------------------------
root@INBOOKX2PLUS:~# sudo lsof -i :8080
COMMAND    PID USER   FD   TYPE  DEVICE SIZE/OFF NODE NAME
node    290516 root   22u  IPv4 3642272      0t0  TCP *:http-alt (LISTEN)
node    290516 root   26u  IPv4 3646369      0t0  TCP localhost:http-alt->localhost:48816 (ESTABLISHED)
node    290516 root   29u  IPv4 3648449      0t0  TCP localhost:http-alt->localhost:48824 (ESTABLISHED)
--------------------------------------------------------------------
root@INBOOKX2PLUS:~# sudo netstat -tulnp | grep 8080
tcp        0      0 0.0.0.0:8080            0.0.0.0:*               LISTEN      290516/node
------------------------------------------------------------------------------------
sudo ufw disable
----------------------------------------
root@INBOOKX2PLUS:/home/webstormProjects/football-fantasy-client# railway status
Project: genuine-miracle
Environment: production
Service: football-fantasy
root@INBOOKX2PLUS:/home/webstormProjects/football-fantasy-client# railway variables
╔═══════════════════════ Variables for football-fantasy ═══════════════════════╗
║ DATABASE_PRIVATE_URL                   │ postgresql://                       ║
║                                        │ postgres:NEDWzmcvaXTaBUqAAwzlIUsZqh ║
║                                        │ PqZXlM@postgres.railway.internal:54 ║
║                                        │ 32/railway                          ║
║──────────────────────────────────────────────────────────────────────────────║
║ RAILWAY_ENVIRONMENT                    │ production                          ║
║──────────────────────────────────────────────────────────────────────────────║
║ RAILWAY_ENVIRONMENT_ID                 │ 139b677a-b1de-4e0d-9521-            ║
║                                        │ 4ba53d15ada1                        ║
║──────────────────────────────────────────────────────────────────────────────║
║ RAILWAY_ENVIRONMENT_NAME               │ production                          ║
║──────────────────────────────────────────────────────────────────────────────║
║ RAILWAY_PRIVATE_DOMAIN                 │ football-fantasy-                   ║
║                                        │ devops.railway.internal             ║
║──────────────────────────────────────────────────────────────────────────────║
║ RAILWAY_PROJECT_ID                     │ 834526c6-819e-413d-beab-            ║
║                                        │ 2c808875fca5                        ║
║──────────────────────────────────────────────────────────────────────────────║
║ RAILWAY_PROJECT_NAME                   │ genuine-miracle                     ║
║──────────────────────────────────────────────────────────────────────────────║
║ RAILWAY_PUBLIC_DOMAIN                  │ football-fantasy-                   ║
║                                        │ production.up.railway.app           ║
║──────────────────────────────────────────────────────────────────────────────║
║ RAILWAY_SERVICE_BACKEND_RUBY_RAILS_URL │ backend-ruby-rails-                 ║
║                                        │ production.up.railway.app           ║
║──────────────────────────────────────────────────────────────────────────────║
║ RAILWAY_SERVICE_FOOTBALL_FANTASY_URL   │ football-fantasy-                   ║
║                                        │ production.up.railway.app           ║
║──────────────────────────────────────────────────────────────────────────────║
║ RAILWAY_SERVICE_ID                     │ 2ba8e848-b48a-4b15-9e0e-            ║
║                                        │ 08b169a61a6c                        ║
║──────────────────────────────────────────────────────────────────────────────║
║ RAILWAY_SERVICE_NAME                   │ football-fantasy                    ║
║──────────────────────────────────────────────────────────────────────────────║
║ RAILWAY_STATIC_URL                     │ football-fantasy-                   ║
║                                        │ production.up.railway.app           ║
╚══════════════════════════════════════════════════════════════════════════════╝

---------------------------------------------
====================
Starting Healthcheck
====================
Path: /ping
Retry window: 1m40s
 
[1/1] Healthcheck succeeded!
-----------------------------------------------
Starting Container
npm WARN config production Use `--omit=dev` instead.
 INFO  Accepting connections at http://localhost:8080/
 HTTP  7/9/2024 7:39:44 AM 100.64.0.2 GET /ping
 HTTP  7/9/2024 7:39:44 AM 100.64.0.2 Returned 200 in 22 ms

**Comment by Akash1134** [comment]
Hi @Or3yn ,

Your discussion doesn't seem to be related to GitHub Codespaces but rather focuses on deployment issues. You are more likely to get a useful response if you are posting your question(s) in the applicable category and are explicit about what your project entails--giving a few more details might help someone give you a nudge in the right direction. I've gone ahead and moved it for you. Good luck!

**Reply by Or3yn** [reply]
Thanks a lot!

**Comment by Or3yn** [comment, marked as answer]
I just switched to docker container to deploy. That's it ... And worked.
